adapter/utils/src/main/java/org/mobilitydata/gtfsvalidator/timeutils/TimeUtilsImpl.java

```diff
@@ -81,43 +84,36 @@ public Integer convertHHMMSSToIntFromNoonOfDayOfService(final String timeAsStrin
     /**
      * This method converts a number of seconds elapsed since noon of say of service to a string formatted as HH:MM:SS.
      *
-     * @param elapsedDurationSinceNoon the number of seconds elapsed since noon of say of service to be converted as a
-     *                                 string formatted as HH:MM:SS.
+     * @param elapsedDurationSinceNoonInSeconds the number of seconds elapsed since noon of day of service to be
+     *                                          converted as a string formatted as HH:MM:SS.
      * @return the human readable string representation of the number of seconds elapsed since noon of day of service.
      * This string is formatted as follows: HH:MM:SS.
      */
-    public String convertIntegerToHMMSS(final Integer elapsedDurationSinceNoon) {
-        if (elapsedDurationSinceNoon != null) {
-            // Determine number of hours elapsed
-            int hourValue = (int) (TimeUnit.SECONDS.toHours(elapsedDurationSinceNoon) + 12);
-            int minuteValue;
-            int secondValue;
-            // elapsedDurationSinceNoon > 0 for times after noon, e.g 14:00PM is represented a +2*3600s=+7200s
-            // elapsedDurationSinceNoon < 0 for times before noon, e.g 11:00AM is represented as -1*3600s=-3600s
-            if (elapsedDurationSinceNoon > 0) {
-                // Determine number of minutes and seconds elapsed
-                minuteValue = (int) (TimeUnit.SECONDS.toMinutes(elapsedDurationSinceNoon) -
-                        TimeUnit.HOURS.toMinutes(TimeUnit.SECONDS.toHours(elapsedDurationSinceNoon)));
-                secondValue = (int) (TimeUnit.SECONDS.toSeconds(elapsedDurationSinceNoon) -
-                        TimeUnit.MINUTES.toSeconds(TimeUnit.SECONDS.toMinutes(elapsedDurationSinceNoon)));
+    public String convertIntegerToHMMSS(final Integer elapsedDurationSinceNoonInSeconds) {
+        if (elapsedDurationSinceNoonInSeconds != null) {
```

Comment:
Instead of wrapping this entire method in the `if`, I'd use an early return and throw an IllegalArgumentException instead (assuming that we shouldn't be getting `null` values at this point):

```java
if (elapsedDurationSinceNoon == null) {
 throw new IllegalArgumentException("elapsedDurationSinceNoonInSeconds cannot be null");
}
```

If we have to guard against `null` values here, then still use the early return, but return `null` instead and update the Javadocs to say it returns `null` if the input is `null`.

Reply:
I just updated my above comment - if you do throw an `IllegalArgumentException`, make sure you add it the method signature so calling code knows it might throw.